torch/utils/data/datapipes/iter/combining.py

```diff
@@ -175,7 +205,11 @@ def get_next_element_by_instance(self, instance_id: int):
                 if self.buffer_size >= 0 and self.leading_ptr > self.buffer_size + self.slowest_ptr:
                     raise BufferError("ForkerIterDataPipe buffer overflow," +
                                       f"buffer size {self.buffer_size} is insufficient.")
-                yield return_val
+
+                if self.copy_fn is not None and instance_id != 0:
+                    yield self.copy_fn(return_val)
```

Comment:
I think it's better to apply `copy_fn` to all instances rather than instances larger than 0.